lib/pluggy_elixir/auth.ex

```diff
@@ -0,0 +1,51 @@
+defmodule PluggyElixir.Auth do
+  @moduledoc """
+  Handle authentication actions.
+  """
+
+  alias PluggyElixir.Config
+  alias PluggyElixir.Error
+
+  @type t :: %__MODULE__{
+          api_key: binary()
+        }
```

Comment:
:nail_care: 

There was a discussion about %__MODULE__ here: https://github.com/christopheradams/elixir_style_guide/issues/22

And after they documented an alternative that can improve readability like this :point_down: 
https://github.com/christopheradams/elixir_style_guide#alias-self-referencing-modules

Reply:
to be honest I prefer using `__MODULE__` since its already a alias. And the cognition is simple, if you read a `__MODULE__` you know it refers to module/file your are reading.

Reply:
In the style guides the use of `__MODULE__` is recommended any way.
![image](https://user-images.githubusercontent.com/17304947/116083375-ec8bce80-a672-11eb-8b67-56de106f1235.png)